openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -53,13 +53,25 @@ def extract_dates_from_course(course):
             # unless that item already has a relative date set
             for _, section, weeks_to_complete in spaced_out_sections(course):
                 items = [section]
+                has_non_ora_scored_content = False
+                section_date_items = []
                 while items:
                     next_item = items.pop()
+                    # TODO: This is pretty gross, and should maybe be configurable in the future,
+                    # especially if we find ourselves needing more exceptions.
                     if next_item.graded:
                         # TODO: Once studio can manually set relative dates,
                         # we would need to manually check for them here
-                        date_items.append((next_item.location, {'due': weeks_to_complete}))
+                        section_date_items.append((next_item.location, {'due': weeks_to_complete}))
+                        has_non_ora_scored_content = (
```

Comment:
Would it be worth just adding a `break` in here once this is True? Might help going through items that we don't need to.

Reply:
It's actually trying to make sure to set dates on all the content (including inherited dates) in edx-when, which is why it doesn't stop.

Reply:
If I'm reading this correctly, if `has_non_ora_scored_content` is True, we don't add any of the `section_date_items` to `date_items` . And since once `has_non_ora_scored_content` is True, it will remain True for the entire for loop, why can't we just break out early? Maybe I'm misreading something about this function.

Reply:
The intended logic is that `section_date_items` gets all graded content in the section, with dates. If `has_non_ora_scored_content` is `True` (meaning, we ever saw a non-ora scored problem in that section), then we add all the graded content dates to edx-when. Because we need to collect all the graded problem dates, I don't think we can ever break out of the loop early.